### DISCUSSION TITLE
Raspberry Pi Pico Error

### DISCUSSION THREAD

**Discussion by BlissIoT** [earlephilhower/arduino-pico#1915] [Q&A]
Hi,
I’m building a soil moisture sensor using Arduino and I got the Raspberry Pi Pico coding to work however, when trying to upload a new one for OLED display I am receiving an error that says, “Error compiling for board Raspberry Pi Pico.” 

I did try reinstalling the Raspberry Pico board but it still doesn’t work.

Here is the full message
“C:\Users\letew\OneDrive\Documents\Arduino\libraries\Adafruit_SSD1306\Adafruit_SSD1306.cpp:42:10: fatal error: pgmspace.h: No such file or directory
 #include <pgmspace.h>
          ^~~~~~~~~~~~
compilation terminated.
exit status 1
Error compiling for board Raspberry Pi Pico.”

![IMG_0549](https://github.com/earlephilhower/arduino-pico/assets/154944484/5632cfee-9e04-4900-baf2-94c77a428f0a)

**Comment by Andy2No** [comment]
We'd need more information to go on.  You can make the console area at the bottom larger (dark bit) by dragging.  You'll then see more details of the error.  Copying that into a text file, and pasting in the relevant parts, works better than a screenshot.

**Reply by BlissIoT** [reply]
Will do, thanks!

**Reply by BlissIoT** [reply]
@Andy2No I have added the full message of the error above.

**Reply by earlephilhower** [reply]
Looking at your photo I can tell you're not using this core, first of all (there are normally ~8 options listed on the bottom status line, not just `RPI Pico on X`).  You're using the Arduino MBED one which does not support that header, evidently.  You should [head over there](https://forum.arduino.cc/) for more assistance since it seems they don't have a `pgmspace.h` header.
